standalone/standalone.py

```diff
@@ -2873,6 +2888,16 @@ def decode_base64(data):
                 # Validate the endpoint
                 endpoint = decode_base64(secret.data.get("endpoint"))
                 validate_url(endpoint)
+
+                # Validate sdg pipeline type constraints
+                if secret.data.get("pipeline_type") and secret.data.get(
```

Comment:
This could be a programming preference so feel free to reject it but I'd personally treat both cases separately:

1. the `pipeline_type` key does not exist
2. the key exists but the value if incorrect

With this we can raise more explicit errors.


```python
if not secret.data.get("pipeline_type"):
    raise ValueError("The provided secret must contain a 'pipeline_type' key.")

if secret.data["pipeline_type"] not in ["simple", "agentic"]:
    raise ValueError(
        f"The provided secret {sdg_serving_model_secret} must contain a valid "
        f"value for 'pipeline_type', one of [simple, agentic]."
    )
```

Reply:
@leseb my understanding from your last comment about making this optional in the secret, means that it is a valid scenario that the this value is not present, so throwing on if not secret.data.get("pipeline_type"): would mean we make this case required, no?

Reply:
Actually, you're right forget about what I said, the current approach is correct, if the value is not present we use the default anyway!